Test.c

```diff
@@ -162,6 +164,52 @@ Test(dci1Test, dci1_calculateShiftOriginTest)
 	{
 		dci1_offsetArray[dci1_bitmap] = possibleLengthBitsRBG[i];
 		testVal = dci1_calculateShiftOrigin(dci1_offsetArray);
-		cr_expect(testVal = correctValues[i], "Value returned from dci1_calculateShiftOrigin [%d] test loop are incorrect", i);
+		cr_expect(testVal = correctValues[i], "Value returned from dci1_calculateShiftOrigin "
+				"[%d] test loop are incorrect", i);
+	}
+}
+
+Test(endToEndTests, dci1_endToEndTest)
+{
+#define TEST_ARR_LEN 4
```

Comment:
if you make a #define only for some part of the file, and not the whole file, please use also #undef at the end of its "scope"